sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -227,48 +247,59 @@ def test_regression_synthetic():
 
     # Friedman1
     X, y = datasets.make_friedman1(n_samples=1200,
-                                   random_state=random_state, noise=1.0)
+                                   random_state=random_state, 
+                                   noise=1.0)
     X_train, y_train = X[:200], y[:200]
     X_test, y_test = X[200:], y[200:]
-    clf = GradientBoostingRegressor()
-    clf.fit(X_train, y_train)
-    mse = mean_squared_error(y_test, clf.predict(X_test))
-    assert mse < 5.0, "Failed on Friedman1 with mse = %.4f" % mse
+
+    for presort in True, False:
+        clf = GradientBoostingRegressor(presort=presort)
+        clf.fit(X_train, y_train)
+        mse = mean_squared_error(y_test, clf.predict(X_test))
+        assert_less(mse, 5.0)
 
     # Friedman2
     X, y = datasets.make_friedman2(n_samples=1200, random_state=random_state)
     X_train, y_train = X[:200], y[:200]
     X_test, y_test = X[200:], y[200:]
-    clf = GradientBoostingRegressor(**regression_params)
-    clf.fit(X_train, y_train)
-    mse = mean_squared_error(y_test, clf.predict(X_test))
-    assert mse < 1700.0, "Failed on Friedman2 with mse = %.4f" % mse
+
+    for presort in True, False:
+        regression_params['presort'] = presort
+        clf = GradientBoostingRegressor(**regression_params)
+        clf.fit(X_train, y_train)
+        mse = mean_squared_error(y_test, clf.predict(X_test))
+        assert_less(mse, 1700.0)
 
     # Friedman3
     X, y = datasets.make_friedman3(n_samples=1200, random_state=random_state)
     X_train, y_train = X[:200], y[:200]
     X_test, y_test = X[200:], y[200:]
-    clf = GradientBoostingRegressor(**regression_params)
-    clf.fit(X_train, y_train)
-    mse = mean_squared_error(y_test, clf.predict(X_test))
-    assert mse < 0.015, "Failed on Friedman3 with mse = %.4f" % mse
+
+    for presort in True, False:
+        regression_params['presort'] = presort
+        clf = GradientBoostingRegressor(**regression_params)
+        clf.fit(X_train, y_train)
+        mse = mean_squared_error(y_test, clf.predict(X_test))
+        assert_less(mse, 0.015)
 
 
 def test_feature_importances():
     X = np.array(boston.data, dtype=np.float32)
     y = np.array(boston.target, dtype=np.float32)
 
-    clf = GradientBoostingRegressor(n_estimators=100, max_depth=5,
-                                    min_samples_split=1, random_state=1)
-    clf.fit(X, y)
-    #feature_importances = clf.feature_importances_
-    assert_true(hasattr(clf, 'feature_importances_'))
+    for presort in True, False:
+        clf = GradientBoostingRegressor(n_estimators=100, max_depth=5,
+                                        min_samples_split=1, random_state=1,
+                                        presort=presort)
+        clf.fit(X, y)
+        #feature_importances = clf.feature_importances_
+        assert_true(hasattr(clf, 'feature_importances_'))
```

Comment:
the commented code `#feature_importances = clf.feature_importances_` is strange.
I would remove the line as it induces me an error.